packages/pipeline-editor/src/PipelineService.tsx

```diff
@@ -370,18 +372,31 @@ export class PipelineService {
 
   static setNodePathsRelativeToWorkspace(
     pipeline: any,
-    pipelinePath: string
+    pipelinePath: string,
+    components: any[]
   ): any {
     for (const node of pipeline.nodes) {
-      if (
-        node.op === 'execute-notebook-node' ||
-        node.op === 'execute-python-node' ||
-        node.op === 'execute-r-node'
-      ) {
-        node.app_data.component_parameters.filename = this.getWorkspaceRelativeNodePath(
-          pipelinePath,
-          node.app_data.component_parameters.filename
-        );
+      const nodeDef = components.find((n: any) => {
+        return n.op === node.op;
+      });
+      const params = nodeDef?.app_data?.properties?.uihints?.parameter_info;
+      if (!params) {
+        continue;
+      }
+      for (const param of params) {
+        if (
+          param.data?.format === 'file' &&
+          param.custom_control_id === 'StringControl'
```

Comment:
Why do we need to check this as well? is there a case where the format is `file` but it's not a string?

Reply:
I don't think that we should check either of these. I would say this is a case where we only would update the path of the property that the `filehandler` parameter ref is pointing to

Reply:
I realize the papermill op doesn't have `filehandler` though

Reply:
@ajbozarth we have to check `StringControl` because `StringArrayControl` can also have `format: "file"` and we don't want to convert any of those

Reply:
@marthacryan can we add a comment explaining this?

Reply:
Closing this thread since I now understand the purpose